### ISSUE TITLE
Broken indexing?

### ISSUE THREAD

**Issue by sarahwie** [pytorch/pytorch#17237]
## 🐛 Bug

Python crashes when attempt to index a tensor. Cannot reproduce in python2.7.

## To Reproduce

Minimal example:
```
import torch
store = torch.randn((12,100,6,50))
store[:, :, 0, :]
```
produces
```
dyld: lazy symbol binding failed: Symbol not found: _PySlice_Unpack
  Referenced from: /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/torch/lib/libtorch_python.dylib
  Expected in: flat namespace

dyld: Symbol not found: _PySlice_Unpack
  Referenced from: /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/torch/lib/libtorch_python.dylib
  Expected in: flat namespace

Abort trap: 6
```

## Environment

PyTorch version: 1.0.1.post2
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.13.6
GCC version: Could not collect
CMake version: version 3.1.2

Python version: 3.6
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] numpy==1.16.1
[pip3] torch==1.0.1.post2
[conda] Could not collect

## Additional Info

Installed using `pip3 install torch`

**Comment by sarahwie** [comment]
Resolved by upgrading to >= Python 3.6.1

**Comment by soumith** [comment]
this is a known issue with python 3.6.1, if my memory serves me right. see https://bugzilla.redhat.com/show_bug.cgi?id=1435135


Upgrade to python 3.6.2 or greater.
(the issue has nothing to do with pytorch and is an abi incompatibility issue across 3.6.0 / 3.6.1)
